src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String contactSegment, boolean isPrivate){
+    	this.value = contactSegment.trim();
+    	this.isPrivate = isPrivate;
+    }
+    
+    /**
+     * Returns true if a given string is a valid person email.
+     */
+    public static boolean isValidSegment(String test, String validationRegex) {
```

Comment:
good job extracting this method and reusing it by taking in validationRegex, which can be supplied by its subclasses.